src/pkg/packager/lint/lint_test.go

```diff
@@ -71,31 +66,75 @@ func TestValidateSchema(t *testing.T) {
+		require.Equal(t, "Additional property not-path is not allowed", validator.findings[0].String())
+		require.Equal(t, "Invalid type. Expected: string, given: integer", validator.findings[1].String())
 	})
 
 	t.Run("Template in component import success", func(t *testing.T) {
 		unmarshalledYaml := readAndUnmarshalYaml[types.ZarfPackage](t, goodZarfPackage)
 		validator := Validator{typedZarfPackage: unmarshalledYaml}
-		checkForVarInComponentImport(&validator)
-		require.Empty(t, validator.warnings)
+		lintComponents(&validator)
+		require.Empty(t, validator.findings)
 	})
 
 	t.Run("Template in component import failure", func(t *testing.T) {
-		unmarshalledYaml := readAndUnmarshalYaml[types.ZarfPackage](t, badZarfPackage)
-		validator := Validator{typedZarfPackage: unmarshalledYaml}
-		checkForVarInComponentImport(&validator)
-		require.Equal(t, validator.warnings[0], ".components.[2].import.path: Will not resolve ZARF_PKG_TMPL_* variables")
-		require.Equal(t, validator.warnings[1], ".components.[3].import.url: Will not resolve ZARF_PKG_TMPL_* variables")
+		pathVar := "###ZARF_PKG_TMPL_PATH###"
+		ociPathVar := fmt.Sprintf("oci://%s", pathVar)
+		pathComponent := types.ZarfComponent{Import: types.ZarfComponentImport{Path: pathVar}}
+		URLComponent := types.ZarfComponent{Import: types.ZarfComponentImport{URL: ociPathVar}}
+		validator := Validator{typedZarfPackage: types.ZarfPackage{Components: []types.ZarfComponent{pathComponent, URLComponent}}}
+		lintUnEvaledVariables(&validator)
+		require.Equal(t, pathVar, validator.findings[0].item)
+		require.Equal(t, ociPathVar, validator.findings[1].item)
+	})
+
+	t.Run("Unpinnned repo warning", func(t *testing.T) {
+		validator := Validator{}
+		unpinnedRepo := "https://github.com/defenseunicorns/zarf-public-test.git"
+		component := types.ZarfComponent{Repos: []string{
+			unpinnedRepo,
+			"https://dev.azure.com/defenseunicorns/zarf-public-test/_git/zarf-public-test@v0.0.1"}}
+		checkForUnpinnedRepos(&validator, 0, component, packageKey{})
+		require.Equal(t, unpinnedRepo, validator.findings[0].item)
+		require.Equal(t, len(validator.findings), 1)
+	})
+
+	t.Run("Unpinnned image warning", func(t *testing.T) {
+		validator := Validator{}
+		unpinnedImage := "registry.com:9001/whatever/image:1.0.0"
+		badImage := "badimage:badimage@@sha256:3fbc632167424a6d997e74f5"
+		component := types.ZarfComponent{Images: []string{
+			unpinnedImage,
+			"busybox:latest@sha256:3fbc632167424a6d997e74f52b878d7cc478225cffac6bc977eedfe51c7f4e79",
+			badImage}}
+		checkForUnpinnedImages(&validator, 0, component, packageKey{})
+		require.Equal(t, unpinnedImage, validator.findings[0].item)
+		require.Equal(t, badImage, validator.findings[1].item)
+		require.Equal(t, 2, len(validator.findings))
+
+	})
+
+	t.Run("Unpinnned file warning", func(t *testing.T) {
+		validator := Validator{}
+		filename := "http://example.com/file.zip"
+		zarfFiles := []types.ZarfFile{
+			{
+				Source: filename,
+			},
+		}
```

Comment:
We should probably have some control options so that we make sure we only have one finding (since a pin and a local shouldn't have findings)